samples/web/content/apprtc/join_page_test.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+
+import unittest
+import webtest
+
+import apprtc
+import constants
+import gcmrecord
+import json
+from google.appengine.datastore import datastore_stub_util
+from google.appengine.api import memcache
+from google.appengine.ext import testbed
+
+class AppRtcPageHandlerTest(unittest.TestCase):
+  def setUp(self):
+    # First, create an instance of the Testbed class.
+    self.testbed = testbed.Testbed()
+
+    # Then activate the testbed, which prepares the service stubs for use.
+    self.testbed.activate()
+
+    # Create a consistency policy that will simulate the High Replication
+    # consistency model.
+    self.policy = datastore_stub_util.PseudoRandomHRConsistencyPolicy()
+    # Initialize the datastore stub with this policy.
+    self.testbed.init_datastore_v3_stub(consistency_policy=self.policy)
+    # Next, declare which service stubs you want to use.
+    self.testbed.init_memcache_stub()
+
+    self.test_app = webtest.TestApp(apprtc.app)
+
+  def tearDown(self):
+    self.testbed.deactivate()
+
+  def makeGetRequest(self, path):
+    # PhantomJS uses WebKit, so Safari is closest to the thruth.
+    return self.test_app.get(path, headers={'User-Agent': 'Safari'})
+
+  def makePostRequest(self, path, body=''):
+    return self.test_app.post(path, body, headers={'User-Agent': 'Safari'})
+    
+  def addTestData(self):
+    gcmrecord.GCMRecord.add_or_update('caller1', 'caller1gcm1', 'caller1code1')
```

Comment:
Put the data in lists and make the calls in a loop, to make the code more concise?